src/libraries/System.Private.Xml/src/System/Xml/Cache/XPathDocumentNavigator.cs

```diff
@@ -113,10 +117,12 @@ public override string Value
                 while (XPathNodeHelper.GetTextFollowing(ref page, ref idx, pageEnd, idxEnd))
                 {
                     Debug.Assert(page[idx].NodeType == XPathNodeType.Element || page[idx].IsText);
+                    value = page[idx].Value;
+                    Debug.Assert(value != null);
```

Comment:
should you just allow null here? I think Append works fine with null

Reply:
This is also set to `s` which is non-nullable.